plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -88,12 +89,17 @@ import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.CodewhispererCompletionType
 import software.aws.toolkits.telemetry.CodewhispererSuggestionState
 import software.aws.toolkits.telemetry.CodewhispererTriggerType
+import java.util.concurrent.CancellationException
 import java.util.concurrent.TimeUnit
 
 @Service
 class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
     private val codeInsightSettingsFacade = CodeInsightsSettingsFacade()
     private var refreshFailure: Int = 0
+    private val ongoingRequests = mutableMapOf<Int, InvocationContext?>()
+    val ongoingRequestsContext = mutableMapOf<Int, RequestContext>()
+    private var jobId = 0
+    private var sessionContext: SessionContext? = null
```

Comment:
not sure do we need this to be thread safe

Reply:
Technically it needs, so the current rule is to modify this only in EDT so I believe in various places I have `@RequiresEDT` annotation including `disposeDisplaySession`, and initialing SessionContext also is in EDT.